### PR TITLE
SSO buttons operational on new signup flow

### PR DESCRIPTION
**Pull request by hannahbergam**
I was able to get the SSO buttons working and test them locally. With google you need to use the 127.0.0.1 url to get it working, and it currently redirects back to 'finish_sign_up' which should be rerouted in our save user pr. 

These only work if you have client keys in your locals file. Additionally, microsoft needs https_development: true in that same file to work (and then with the url https://localhost-studio.code.org:3000/users/new_sign_up). My mac was having [trouble getting https working locally](https://codedotorg.slack.com/archives/C046G4TRLEN/p1721137347331789) (and sso is notoriously trickly locally). Additionally, Turner and Kelby were not able to get secure localhost running after a few hours of attempts, so I would prefer to try the other three on the 'hiding in plain sight' page in production to see if I get any errors. 

I was also able to get the clever image next to the button because our design system Button is incompatible with a form (type='submit') so I had to start over with all the buttons and icons.

Screenshot 2024-08-22 at 2 42 57 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2265, https://codedotorg.atlassian.net/browse/ACQ-2264

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/signUpFlow/loginTypeSelection.module.scss [hunk 1]
```diff
@@ -113,6 +117,7 @@ body {
     .googleButton {
       background-color: $google_brand_color;
       margin-bottom: .5rem;
+      color: $white;
```
Comment (kelbyhawn) [inline apps/src/signUpFlow/loginTypeSelection.module.scss]: You might be able to refactor this to group shared button styles using the `button` selector (or add a second repeating class to each button) for things like margin, text color, font-weight etc. Or even put them in a container and use flexbox for the spacing between them.

**Review by kelbyhawn — COMMENTED** [review]
I have a couple small ui nits, but otherwise these look awesome! Happy to pair if you want an extra set of eyes.
- The font size and weight should match the Create my account button (even though the font weight on both buttons in the Figma looks thinner)
- The button padding should match the Create my account button so they're the same height; these look a little too short, but maybe it's the font-weight causing an optical illusion 🪄

File: apps/src/signUpFlow/loginTypeSelection.module.scss (hunk 1 above)
Comment (hannahbergam) [reply apps/src/signUpFlow/loginTypeSelection.module.scss]: Yes! Is it alright if I wait for Turner's big refactor to do more specific style cleanup?

Comment (kelbyhawn) [reply apps/src/signUpFlow/loginTypeSelection.module.scss]: Oh yes absolutely! I can also do a final css pass before we launch 🙌

**Review by kelbyhawn — APPROVED** [review]
🎉

**Comment by hannahbergam** [comment]
> 🎉

Hi... I had to do some sad updates to the styles (i.e. remove the usage of 'gap') because we no longer get to work with the nice design component buttons and have to work with weird 'forms' and 'button' components instead. I did what I had to to get it presentable enough for now and will leave the rest for a styling cleanup. It's also worth noting that not all of the buttons have all the same properties (some are white, one is black, some use font awesome icons, another uses an image), so the abstraction there will be a little messy too.

**Comment by kelbyhawn** [comment]
> Hi... I had to do some sad updates to the styles…

That's ok! We'll get it looking perfect before it launches.

**Pull request merged**